common/app/routes/challenges/redux/reducer.js

```diff
@@ -64,6 +65,16 @@ const initialState = {
 
 const mainReducer = handleActions(
   {
+    [types.isPassed]: state => {
+      let { tests } = state;
+    if (tests.length > 0 && tests.every(test => test.pass && !test.err)) {
+      return ({
+              ...state,
+              isChallengePassed: true
+              });
+    }
+    return state;
```

Comment:
There is no way to unset isChallengePassed. If the user changes their code, it should reflect that the challenge no longer is passed.